sdks/java/core/src/main/java/org/apache/beam/sdk/sd/BeamRowType.java

```diff
@@ -15,23 +15,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.beam.dsls.sql.schema;
+package org.apache.beam.sdk.sd;
 
 import com.google.auto.value.AutoValue;
 import java.io.Serializable;
 import java.util.List;
 
 /**
- * Field type information in {@link BeamSqlRow}.
+ * Field type information in {@link BeamRow}.
  *
  */
 @AutoValue
-public abstract class BeamSqlRowType implements Serializable {
+public abstract class BeamRowType implements Serializable {
   public abstract List<String> getFieldsName();
   public abstract List<Integer> getFieldsType();
 
-  public static BeamSqlRowType create(List<String> fieldNames, List<Integer> fieldTypes) {
-    return new AutoValue_BeamSqlRowType(fieldNames, fieldTypes);
+  public static BeamRowType create(List<String> fieldNames, List<Integer> fieldTypes) {
```

Comment:
This interface (taking a list of names, then a parallel list of types) seems fragile and prone to off-by-one errors. If columns are always referred to by name (not index) in the API, could this just be a Map? Otherwise, some kind of an OrderedMap (or list of pairs)?